### PR TITLE
feat(spring-sdk): spring based integration testkit

### PR DESCRIPTION
**Pull request by octonato**
References https://github.com/lightbend/kalix/issues/7440

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -80,6 +81,30 @@ lazy val sdkJava = project
...
+    buildInfoPackage := "kalix.javasdk.testkit",
+    // Generate javadocs by just including non generated Java/Scala sources
+    Compile / doc / sources := {
+      val javaSourceDir = (Compile / javaSource).value.getAbsolutePath
+      val scalaSourceDir = (Compile / scalaSource).value.getAbsolutePath
+      (Compile / doc / sources).value.filter(f =>
+        f.getAbsolutePath.startsWith(javaSourceDir) || f.getAbsolutePath.startsWith(scalaSourceDir))
+    })
+  .settings(Dependencies.sdkJavaTestKit)
+
```
Comment (octonato) [inline build.sbt:107]: just moving it to close to the sdkJava

File: samples/spring-customer-registry-views-quickstart/src/it/java/customer/api/CustomerIntegrationTest.java [hunk 2]
```diff
@@ -2,34 +2,29 @@
...
 
+import static java.time.temporal.ChronoUnit.SECONDS;
 import static org.awaitility.Awaitility.await;
 
 @RunWith(SpringRunner.class)
 @SpringBootTest(classes = Main.class)
-@Import(KalixConfigurationTest.class)
-@TestPropertySource(properties = "spring.main.allow-bean-definition-overriding=true")
-public class CustomerIntegrationTest {
+public class CustomerIntegrationTest extends KalixIntegrationTestKitSupport {
```
Comment (octonato) [inline samples/spring-customer-registry-views-quickstart/src/it/java/customer/api/CustomerIntegrationTest.java:27]: At the end, the path with less burden to the user is to extends this class. There are a few configs in place that we don't want to ask users to copy over and over. Easier to say, extends this and everything will work (hopefully).

File: sdk/spring-sdk-testkit/src/main/java/kalix/springsdk/KalixIntegrationTestKitSupport.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * <p>Users can interact their components using the {@link
+ * org.springframework.web.reactive.function.client.WebClient} that is made available by the test
+ * {@link org.springframework.context.ApplicationContext}.
+ *
+ * <p>On test teardown, the Kalix server and the Kalix proxy (docker container) will be stopped.
+ */
+@Import(KalixConfigurationTest.class)
+@TestPropertySource(properties = "spring.main.allow-bean-definition-overriding=true")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Comment (octonato) [inline sdk/spring-sdk-testkit/src/main/java/kalix/springsdk/KalixIntegrationTestKitSupport.java:44]: we need this config to: 

* override a bean defined for prod, ie: KalixServer 
* be able to run a `@AfterAll` and stop the testkit.

File: sdk/spring-sdk-testkit/src/main/java/kalix/springsdk/KalixIntegrationTestKitSupport.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import org.junit.jupiter.api.TestInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.TestPropertySource;
+
+/**
+ * This class provided the necessary infrastructure to run Kalix integration test for projects built
+ * with the Spring SDK. Users should let their tests classes extends this class.
```
Comment (efgpinto) [inline sdk/spring-sdk-testkit/src/main/java/kalix/springsdk/KalixIntegrationTestKitSupport.java:30]:
```suggestion
 * with the Spring SDK. Users should let their test classes extends this class.
```

File: sdk/spring-sdk-testkit/src/main/java/kalix/springsdk/KalixIntegrationTestKitSupport.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * <p>Requires Docker for starting a local instance of the Kalix proxy.
+ *
+ * <p>This class wires-up a local Kalix server using the user's defined Kalix components.
+ *
+ * <p>Users can interact their components using the {@link
```
Comment (efgpinto) [inline sdk/spring-sdk-testkit/src/main/java/kalix/springsdk/KalixIntegrationTestKitSupport.java:36]:
```suggestion
 * <p>Users can interact with their components using the {@link
```

**Review by efgpinto — APPROVED** [review]
LGTM.. found 2 small typos

**Comment by octonato** [comment]
Thanks @efgpinto. I applied the typo fixes. Will merge when green.

**Pull request merged**